File: model/src/main/java/io/github/runtimemodels/chazm/relation/ModeratesFactory.java
package io.github.runtimemodels.chazm.relation;

import io.github.runtimemodels.chazm.entity.Attribute;
import io.github.runtimemodels.chazm.entity.Pmf;

/**
 * The {@linkplain ModeratesFactory} interface defines the API for constructing {@linkplain Moderates} relations.
 *
 * @author Christopher Zhong
 * @since 7.0.0
 */
@FunctionalInterface
public interface ModeratesFactory {

    /**
     * Constructs a {@linkplain Moderates}.
     *
     * @param pmf       the {@linkplain Pmf} of the {@linkplain Moderates}.
     * @param attribute the {@linkplain Attribute} of the {@linkplain Moderates}.
     * @return a {@linkplain Moderates}.
     */
    Moderates buildModerates(Pmf pmf, Attribute attribute);

}

File: chazm-model/src/main/kotlin/runtimemodels/chazm/model/relation/ModeratesFactory.kt
package runtimemodels.chazm.model.relation

import runtimemodels.chazm.api.entity.Attribute
import runtimemodels.chazm.api.entity.Pmf
import runtimemodels.chazm.api.relation.Moderates

/**
 * The [ModeratesFactory] interface defines the API for constructing [Moderates] relations.
 *
 * @author Christopher Zhong
 * @since 7.0.0
 */
@FunctionalInterface
interface ModeratesFactory {
    /**
     * Constructs a [Moderates].
     *
     * @param pmf       the [Pmf] of the [Moderates].
     * @param attribute the [Attribute] of the [Moderates].
     * @return a [Moderates].
     */
    fun build(pmf: Pmf, attribute: Attribute): Moderates
}
